// hexo-theme-shokax/source/js/_app/globals/globalVars.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setSiteSearch = exports.setToolBtn = exports.setShowContents = exports.setGoToComment = exports.setBackToTop = exports.setToolPlayer = exports.setOriginTitle = exports.setPjax = exports.setLocalUrl = exports.setLocalHash = exports.setTitleTime = exports.setDiffY = exports.setOWinWidth = exports.setOWinHeight = exports.setHeaderHight = exports.setHeaderHightInner = exports.setSiteNavHeight = exports.pjax = exports.LOCAL_URL = exports.LOCAL_HASH = exports.oWinWidth = exports.oWinHeight = exports.headerHight = exports.headerHightInner = exports.siteNavHeight = exports.siteSearch = exports.showContents = exports.goToComment = exports.backToTop = exports.toolPlayer = exports.toolBtn = exports.siteBrand = exports.sideBar = exports.quickBtn = exports.menuToggle = exports.siteHeader = exports.siteNav = exports.loadCat = exports.Container = exports.HTML = exports.BODY = exports.titleTime = exports.originTitle = exports.diffY = exports.scrollAction = exports.statics = void 0;
const dom_1 = require("../library/dom");
const proto_1 = __importDefault(require("../library/proto"));
(0, proto_1.default)();
exports.statics = CONFIG.statics.indexOf('//') > 0 ? CONFIG.statics : CONFIG.root;
exports.scrollAction = { x: 0, y: 0 };
exports.diffY = 0;
exports.BODY = document.getElementsByTagName('body')[0];
exports.HTML = document.documentElement;
exports.Container = (0, dom_1.$dom)('#container');
exports.loadCat = (0, dom_1.$dom)('#loading');
exports.siteNav = (0, dom_1.$dom)('#nav');
exports.siteHeader = (0, dom_1.$dom)('#header');
exports.menuToggle = exports.siteNav.child('.toggle');
exports.quickBtn = (0, dom_1.$dom)('#quick');
exports.sideBar = (0, dom_1.$dom)('#sidebar');
exports.siteBrand = (0, dom_1.$dom)('#brand');
exports.toolBtn = (0, dom_1.$dom)('#tool');
exports.siteSearch = (0, dom_1.$dom)('#search');
exports.oWinHeight = window.innerHeight;
exports.oWinWidth = window.innerWidth;
exports.LOCAL_HASH = 0;
exports.LOCAL_URL = window.location.href;
function setSiteNavHeight(value) {
    exports.siteNavHeight = value;
}
exports.setSiteNavHeight = setSiteNavHeight;
function setHeaderHightInner(value) {
    exports.headerHightInner = value;
}
exports.setHeaderHightInner = setHeaderHightInner;
function setHeaderHight(value) {
    exports.headerHight = value;
}
exports.setHeaderHight = setHeaderHight;
function setOWinHeight(value) {
    exports.oWinHeight = value;
}
exports.setOWinHeight = setOWinHeight;
function setOWinWidth(value) {
    exports.oWinWidth = value;
}
exports.setOWinWidth = setOWinWidth;
function setDiffY(value) {
    exports.diffY = value;
}
exports.setDiffY = setDiffY;
function setTitleTime(value) {
    exports.titleTime = value;
}
exports.setTitleTime = setTitleTime;
function setLocalHash(value) {
    exports.LOCAL_HASH = value;
}
exports.setLocalHash = setLocalHash;
function setLocalUrl(value) {
    exports.LOCAL_URL = value;
}
exports.setLocalUrl = setLocalUrl;
function setPjax(value) {
    exports.pjax = value;
}
exports.setPjax = setPjax;
function setOriginTitle(value) {
    exports.originTitle = value;
}
exports.setOriginTitle = setOriginTitle;
function setToolPlayer(value) {
    exports.toolPlayer = value;
}
exports.setToolPlayer = setToolPlayer;
function setBackToTop(value) {
    exports.backToTop = value;
}
exports.setBackToTop = setBackToTop;
function setGoToComment(value) {
    exports.goToComment = value;
}
exports.setGoToComment = setGoToComment;
function setShowContents(value) {
    exports.showContents = value;
}
exports.setShowContents = setShowContents;
function setToolBtn(value) {
    exports.toolBtn = value;
}
exports.setToolBtn = setToolBtn;
function setSiteSearch(value) {
    exports.siteSearch = value;
}
exports.setSiteSearch = setSiteSearch;
